tests/integration/ovs_test.py

```diff
@@ -18,103 +18,90 @@
-    }
-    libnmstate.apply(desired_state)
+def test_create_and_remove_ovs_bridge_options_specified():
+    bridge = create_bridge_subtree_state(
+        {
+            OVSBridge.FAIL_MODE: '',
+            OVSBridge.MCAST_SNOOPING_ENABLED: False,
+            OVSBridge.RSTP: False,
+            OVSBridge.STP: True,
+        }
+    )
 
-    setup_remove_ovs_bridge_state = {
-        INTERFACES: [
-            {'name': 'ovs-br0', 'type': 'ovs-bridge', 'state': 'absent'}
-        ]
-    }
-    libnmstate.apply(setup_remove_ovs_bridge_state)
-    state = statelib.show_only((desired_state[INTERFACES][0]['name'],))
-    assert not state[INTERFACES]
+    with ovs_bridge(BRIDGE1, bridge) as bridge_state:
+        assertlib.assert_state_match(bridge_state)
+
+    assertlib.assert_absent(BRIDGE1)
+
+
+@pytest.mark.xfail(
+    raises=NmstateLibnmError, reason='https://bugzilla.redhat.com/1724901'
+)
+def test_create_and_remove_ovs_bridge_with_a_system_port(port0_up):
+    port0_name = port0_up[Interface.KEY][0][Interface.NAME]
+
+    bridge = create_bridge_subtree_state()
+    add_port_to_bridge(bridge, port0_name, OVSBridgePortType.SYSTEM)
+
+    with ovs_bridge(BRIDGE1, bridge) as bridge_state:
+        assertlib.assert_state_match(bridge_state)
+
+    assertlib.assert_absent(BRIDGE1)
 
 
 @pytest.mark.xfail(
     raises=NmstateLibnmError, reason='https://bugzilla.redhat.com/1724901'
 )
-def test_create_and_remove_ovs_bridge_with_an_internal_port():
-    state = yaml.load(OVS_BRIDGE_YAML_BASE, Loader=yaml.SafeLoader)
-    state[INTERFACES][0]['bridge']['port'] = [
-        {'name': 'ovs0', 'type': 'internal'}
-    ]
-    ovs_internal_interface_state = {
-        'name': 'ovs0',
-        'type': 'ovs-interface',
-        'state': 'up',
-        'mtu': 1500,
-        'ipv4': {
+def test_create_and_remove_ovs_bridge_with_internal_port_and_static_ip():
+    bridge = create_bridge_subtree_state()
+    add_port_to_bridge(bridge, PORT1, OVSBridgePortType.INTERNAL)
+
+    ipv4_state = {
+        Interface.IPV4: {
             InterfaceIPv4.ENABLED: True,
             InterfaceIPv4.ADDRESS: [
                 {
                     InterfaceIPv4.ADDRESS_IP: '192.0.2.1',
                     InterfaceIPv4.ADDRESS_PREFIX_LENGTH: 24,
                 }
             ],
-        },
-    }
-    state[INTERFACES].append(ovs_internal_interface_state)
-    libnmstate.apply(state, verify_change=False)
-
-    setup_remove_ovs_bridge_state_and_port = {
-        INTERFACES: [
-            {'name': 'ovs-br0', 'type': 'ovs-bridge', 'state': 'absent'},
-            {'name': 'ovs', 'type': 'ovs-interface', 'state': 'absent'},
-        ]
+        }
     }
-    libnmstate.apply(setup_remove_ovs_bridge_state_and_port)
-    state = statelib.show_only(
-        (state[INTERFACES][0]['name'], state[INTERFACES][1]['name'])
-    )
-    assert not state[INTERFACES]
+
+    with ovs_interface(PORT1, ipv4_state) as iface_state:
```

Comment:
It does not seem to make a lot of sense to create it in a separate transaction. Consider using the ‘create’ flag as in  https://github.com/nmstate/nmstate/blob/bb9b06287d1e5f04c74c892a1aea14afcc484ab3/tests/integration/testlib/bridgelib.py#L49

Reply:
Considered, but I prefer it this way. If I use `create=False` I need to add join the configuration and then apply it myself and it lowers readability of the test.

Reply:
On the other hand, the test now is highly dependent on the NM provider.
This would have never worked with an ovsdb or ovs-vsctl provider.
An internal interface cannot exist without a bridge. I think it is wrong for NM to represent the interface as `UP` (active) without it being actually created.

Reply:
If you want me to change this to create=False, just say a word. I'd prefer not to theorize about corner-cases of a driver that may or may not appear in the future. Should I open a bug on NM/nmstate?

Reply:
I am asking to do this setup in one transaction, especially when they are so coupled anyway.
If you use `create` or something else, it is up to you.

And yes, I think you should do the minimum to be generic and not to depend on a specific implementation. Unless it is too complicated.

Regarding NM, I would check with the NM team before opening the BZ, but I tend to think it is a bug.

Reply:
Done